src/allmydata/test/web/test_root.py

```diff
@@ -36,3 +45,363 @@ def test_missing(self):
         self.assertIdentical(res, ctx)
         self.assertEqual(ctx.slots["version"], "")
         self.assertEqual(ctx.slots["nickname"], "")
+
+
+class FakeUploader(object):
+    """
+    """
+    def get_helper_info(self):
+        return ("furl", False)
+
+
+class FakeHelper(object):
+    def get_stats(self):
+        return {
+            "chk_upload_helper.active_uploads": 0,
+        }
+
+
+class FakeStorageBroker(object):
+    def get_connected_servers(self):
+        return {}
+
+    def get_all_serverids(self):
+        return ()
+
+    def get_known_servers(self):
+        return ()
+
+
+class FakeMagicFolder(object):
+    def get_public_status(self):
+        return (True, ["this magic folder is alive"])
+
+
+# XXX there are several 'fake client' instance throughout the code
+# .. probably should be a single one that works better and covers all
+# the cases etc.
```

Comment:
Oh.  Yea.  Look at that. :)

Do you want to tackle this here?  The job is only going to get larger...

Reply:
Yeah, not sure if "here, and now" is the place? This is always my problem with Tahoe PRs ;) -- that I want to fix more and more things. So I placate myself with `XXX` comments...

Reply:
Maybe this could turn into a ticket reference.

Reply:
Filed https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3253 and deleted these tests